apis/v1beta1/providerconfig_types.go

```diff
@@ -27,6 +27,12 @@ type ProviderConfigSpec struct {
 	// Credentials required to authenticate to this provider.
 	Credentials ProviderCredentials `json:"credentials"`
 
+	// AssumeRoleARN to assume with provider credentials
+	// can be used together with InjectedIdentity
```

Comment:
Now that you can assume role with all types of creds, this comment needs to be updated.